src/code/z_parameter.c

```diff
@@ -2532,90 +2562,91 @@ void Interface_UpdateMagicBar(GlobalContext* globalCtx) {
             }
             break;
 
-        case 10:
+        case MAGIC_STATE_ADD:
+            // Add magic until target is reached
             gSaveContext.magic += 4;
             Audio_PlaySoundGeneral(NA_SE_SY_GAUGE_UP - SFX_FLAG, &gSfxDefaultPos, 4, &gSfxDefaultFreqAndVolScale,
                                    &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
-            if (gSaveContext.magic >= gSaveContext.unk_13F8) {
-                gSaveContext.magic = gSaveContext.unk_13F8;
-                gSaveContext.unk_13F0 = gSaveContext.unk_13F2;
-                gSaveContext.unk_13F2 = 0;
+            if (gSaveContext.magic >= gSaveContext.magicTarget) {
+                gSaveContext.magic = gSaveContext.magicTarget;
+                gSaveContext.magicState = gSaveContext.magicStateStored;
+                gSaveContext.magicStateStored = MAGIC_STATE_IDLE;
             }
             break;
 
         default:
-            gSaveContext.unk_13F0 = 0;
+            gSaveContext.magicState = MAGIC_STATE_IDLE;
             break;
     }
 }
 
-void Interface_DrawMagicBar(GlobalContext* globalCtx) {
+void Magic_Draw(GlobalContext* globalCtx) {
     InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
-    s16 magicBarY;
+    s16 magicMeterY;
 
     OPEN_DISPS(globalCtx->state.gfxCtx, "../z_parameter.c", 2650);
 
     if (gSaveContext.magicLevel != 0) {
         if (gSaveContext.healthCapacity > 0xA0) {
-            magicBarY = R_MAGIC_BAR_LARGE_Y;
+            magicMeterY = R_MAGIC_METER_LARGE_Y; // two rows of hearts
         } else {
-            magicBarY = R_MAGIC_BAR_SMALL_Y;
+            magicMeterY = R_MAGIC_METER_SMALL_Y; // one row of hearts
```

Comment:
Not your changes but,
when I first saw these regs' name I thought it refered to half/double magic

My best suggestion is unfortunately only `R_MAGIC_METER_Y_FEW_HEARTS` and `R_MAGIC_METER_Y_MANY_HEARTS` 😅

Reply:
Yes, I see the confusion and think it should be changed. Maybe something like:
`R_MAGIC_METER_Y_POS_UPPER` (= 34)
`R_MAGIC_METER_Y_POS_LOWER` (= 42)
(Upper is a smaller number because it's coordinates from the top of the screen I'm assuming?)

Reply:
Yeah screen coordinates are 0,0 top left and grow towards bottom right

I don't like upper/lower because it describes the data rather than describing what the data is for. (for example github's green comment button is better described as "reply button" rather than "green box with black text")

Reply:
The UPPER is describing the Y_POS which is what this data is for? Would `R_MAGIC_METER_Y_UPPER_POS` be clearer?